pattern_matcher/pattern_matcher.go

```diff
@@ -0,0 +1,76 @@
+package pattern_matcher
+
+import "github.com/gobwas/glob"
+
+type PatternMatcher struct {
+	include []string
+	exclude []string
+}
+
+// Option is the type for functional options.
+type Option func(*PatternMatcher)
+
+func New(options ...Option) *PatternMatcher {
+	pm := &PatternMatcher{
+		include: []string{"*"},
+		exclude: []string{},
+	}
+
+	for _, option := range options {
+		option(pm)
+	}
+	return pm
+}
+
+func WithInclude(list []string) Option {
```

Comment:
WithReplaceIncludes(values []string) which replaces the existing array of filters with the values passed in.

Reply:
done